streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -874,26 +868,26 @@ public void shouldNotBlockInCloseForZeroDuration() {
 
     @Test
     public void shouldTriggerRecordingOfRocksDBMetricsIfRecordingLevelIsDebug() {
-        PowerMock.mockStatic(Executors.class);
-        final ScheduledExecutorService cleanupSchedule = EasyMock.niceMock(ScheduledExecutorService.class);
+        final MockedStatic<Executors> executorsMockedStatic = mockStatic(Executors.class);
+        final ScheduledExecutorService cleanupSchedule = mock(ScheduledExecutorService.class, withSettings().lenient());
         final ScheduledExecutorService rocksDBMetricsRecordingTriggerThread =
-            EasyMock.mock(ScheduledExecutorService.class);
-        EasyMock.expect(Executors.newSingleThreadScheduledExecutor(
-            anyObject(ThreadFactory.class)
-        )).andReturn(cleanupSchedule);
-        EasyMock.expect(Executors.newSingleThreadScheduledExecutor(
-            anyObject(ThreadFactory.class)
-        )).andReturn(rocksDBMetricsRecordingTriggerThread);
-        EasyMock.expect(rocksDBMetricsRecordingTriggerThread.scheduleAtFixedRate(
-            EasyMock.anyObject(RocksDBMetricsRecordingTrigger.class),
-            EasyMock.eq(0L),
-            EasyMock.eq(1L),
-            EasyMock.eq(TimeUnit.MINUTES)
-        )).andReturn(null);
-        EasyMock.expect(rocksDBMetricsRecordingTriggerThread.shutdownNow()).andReturn(null);
-        PowerMock.replay(Executors.class);
-        PowerMock.replay(rocksDBMetricsRecordingTriggerThread);
-        PowerMock.replay(cleanupSchedule);
+            mock(ScheduledExecutorService.class);
+
+        executorsMockedStatic.when(() -> Executors.newSingleThreadScheduledExecutor(
+            any(ThreadFactory.class)
+        )).thenReturn(cleanupSchedule);
+        executorsMockedStatic.when(() -> Executors.newSingleThreadScheduledExecutor(
```

Comment:
The different `ThreadFactory` should get different `ScheduledExecutorService` (see production code `KafkaStreams`). That was a bug in previous test. However, it was fine since all this test want to do is to make sure `rocksDBMetricsRecordingTriggerThread` is executed.

Could you follow the production code to return different `ScheduledExecutorService`? or add comments to say "this is fine because ..."